fili-core/src/main/java/com/yahoo/bard/webservice/data/config/dimension/DefaultDimensionConfig.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.dimension;
+
+import com.yahoo.bard.webservice.data.config.names.DimensionName;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+
+import java.util.LinkedHashSet;
+import javax.validation.constraints.NotNull;
+
+
+/**
+ * A Dimension instance holds all of the information needed to construct a Dimension.
+ */
+public class DefaultDimensionConfig implements DimensionConfig {
```

Comment:
I thought the purpose is to get rid of redundant methods between different types of `DimensionConfig`. This PR is originated from Will's comment [here](https://git.corp.yahoo.com/Digits/Hippodrome/pull/701) saying 
`SHouldn't this be a common bard class rather than reimplementing it here and in UAD?`.